internal/security/secretstore/vmkencryption.go

```diff
@@ -133,22 +133,21 @@ func (v *VMKEncryption) EncryptInitResponse(initResp *secretstoreclient.InitResp
 		wipeKey(nonce)    // Clear out nonce
 	}
 
-	initResp.EncryptedKeys = newKeys
-	initResp.Nonces = newNonces
-	initResp.Keys = nil       // strings are immutable, must wait for GC
-	initResp.KeysBase64 = nil // strings are immutable, must wait for GC
-	return nil
+	encryptedResponse.EncryptedKeys = newKeys
+	encryptedResponse.Nonces = newNonces
+	return encryptedResponse, nil
```

Comment:
@lenny-intel In the original implementation I mutated the input argument because I didn't want to mess with other JSON that may be been in the initResponse.  Specifically, the new implementation will always strip out the RootToken.  Please copy the original initResponse and only change the two(four) fields that this method deals with.

Reply:
@bnevis-i , K, made copy but no sure what you mean by ` two(four) fields` since it is just two fields.

Reply:
NM, got it.

Reply:
Restored original approach